source/use-cases/sensitive-data-encryption.txt

```diff
@@ -124,7 +126,49 @@ In this step, we generate a local master key to expedite setup of our developmen
           }
 
 
-B. Define a JSON Schema
+.. _fle-create-a-data-encryption-key:
+
+B. Create a Data Encryption Key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this section, we configure the MongoDB driver generate a data
+encryption key using the master key. The data encryption key, used to 
+automatically encrypt and decrypt fields in a document, is stored in a MongoDB 
+collection called the *key vault collection*.
+
+The following configuration values are required by the client to generate a new
+data encryption key:
+
+* The local master key or AWS KMS master key access settings.
+* A MongoDB connection string that authenticates on a running server.
+* The key vault namespace (database and collection).
+
+In this guide, we use the ``encryption`` database and ``__keyvault`` 
+collection as our key vault collection.
+
+The MongoDB driver performs the following to generate the data encryption key:
+
+1.  Creates a data encryption key.
+#.  Inserts a new document into the key vault collection.
+#.  Returns the ``_id`` field of the inserted document as a `UUID <https://en.wikipedia.org/wiki/Universally_unique_identifier>`_.
+
+Follow the steps below to generate a single data encryption key from the
+local master key.
+
+.. note::
+
+  If using AWS KMS to provide the master key, refer to the :ref:`Convert
+  to a Remote Master Key <fle-convert-to-a-remote-master-key>` section for
+  the appropriate settings.
+
+.. include:: /includes/steps/create-data-encryption-key.rst
+
+`Download the source code <https://raw.githubusercontent.com/mongodb/docs-assets/DOCSP-csfle-data-encryption-key/DataEncryptionKeyGenerator.java>`_
+used to generate the data encryption key in this section.
```

Comment:
Seems like this might get lost here at the bottom of the step. Maybe we could combine with the "follow the steps" sentence above the stepfile?

Follow the steps below to generate a single data encryption key from the local master key. You can also find the complete code example on GitHub.

Reply:
Sounds good, will update.